fpdf/fpdf.py

```diff
@@ -78,19 +78,21 @@ class Image:
 from .fonts import CoreFont, CORE_FONTS, FontFace, TTFFont
 from .graphics_state import GraphicsStateMixin
 from .html import HTML2FPDF
+from .image_datastructures import (
```

Comment:
Nice continuation on my (so far rather superficial) start to better encapsulate image handling and unify raster and vector processing.

Your changes will conflict with mine in the upcoming table cell/text region PR, but that can't be helped...

On the other hand, image_parsing.py is not all that big. Is it really worth it to seperate out the data structures?

Reply:
> On the other hand, image_parsing.py is not all that big. Is it really worth it to seperate out the data structures?

Yes, this change was needed to avoid circular imports between `fpdf.image_parsing` & `fpdf.fpdf` modules.

The new `fpdf.image_datastructures` module is used by no less than 5 modules: `fpdf.image_parsing`, `fpdf.output`, `fpdf.fpdf`, `fpdf.svg` and `fpdf.text_region`.
On the other hand, `fpdf.image_parsing` is only used by 3 modules: `fpdf.fpdf`, `fpdf.text_region` and `fpdf.svg` (with an ugly function-local import because I couldn't figure a better way to do it, for now).

In the future, maybe we should initiate a `fpdf/images/` sub-folder to gather all image-related modules?

Reply:
> Yes, this change was needed to avoid circular imports between fpdf.image_parsing & fpdf.fpdf modules.

I think you mean between `fpdf.image_parsing` and `fpdf.svg`, there should not have been a cross-dependency with `fpdf.fpdf`. But I did indeed forget that `fpdf.image_parsing` uses `SVGObject` which makes it tricky.

> In the future, maybe we should initiate a fpdf/images/ sub-folder to gather all image-related modules?

Are you expecting more than two image related source files in the future to justify a subdirectory?
Before we start with subdirectories in general, I'd prefer to establish an overall concept for that (eg. core/utils/graphics/text/layout or something similar). But I really don't think this is urgent, we still have less than 30 source files.

Reply:
> I think you mean between `fpdf.image_parsing` and `fpdf.svg`, there should not have been a cross-dependency with `fpdf.fpdf`. But I did indeed forget that `fpdf.image_parsing` uses `SVGObject` which makes it tricky.

Yes, that's what I meant to write 😅

> Are you expecting more than two image related source files in the future to justify a subdirectory?

I thought about maybe putting `fpdf/svg.py` in it too.

> But I really don't think this is urgent, we still have less than 30 source files.

Agreed